src/classes/CW_CampaignTriggerHandlerTest.cls

```diff
@@ -0,0 +1,266 @@
+    	camp.Recurrence_Type__c = ConstantUtil.CAMPAIGN_RECURRENCE_TYPE_RECURRING;
+    	camp.Brief_Approval_Status__c = 'In Approval';
+    	insert camp;
+    	
+    	Test.startTest();
+    	camp.Brief_Approval_Status__c = 'Rejected';
+    	update camp;
+    	Test.stopTest();
+    	system.assertEquals('Draft',[Select Stage__c from CW_Campaign__c where Id=:camp.Id].Stage__c);
+    	
+    	camp.Brief_Approval_Status__c = 'In Approval';
+    	camp.Cost_Approval_Status__c = 'Approved';
+    	camp.Roadmap_Approval_Status__c = 'Approved';
+    	update camp;
+    	
+    	camp.Brief_Approval_Status__c = 'Rejected';
+    	update camp;
+    	system.assertEquals('Roadmap Approved',[Select Stage__c from CW_Campaign__c where Id=:camp.Id].Stage__c);
+    	
+    	//recall brief for an urgent campaign -> camp stage should revert to draft
+    	camp.Brief_Approval_Status__c = 'In Approval';
+    	camp.Roadmap_Approval_Status__c = 'Not Submitted';
+    	update camp;
+    	
+    	camp.Brief_Approval_Status__c = 'Not Submitted';
+    	update camp;
+    	system.assertEquals('Draft',[Select Stage__c from CW_Campaign__c where Id=:camp.Id].Stage__c);
+    	//recall the brief when roadmap is approved -> camp stage should revert to roadmap approved
+    	camp.Brief_Approval_Status__c = 'In Approval';
+    	camp.Roadmap_Approval_Status__c = 'Approved';
+    	update camp;
+    	
+    	camp.Brief_Approval_Status__c = 'Not Submitted';
+    	update camp;
+    	system.assertEquals('Roadmap Approved',[Select Stage__c from CW_Campaign__c where Id=:camp.Id].Stage__c);
+	}
+	
+	//test if campaign roadmap can be removed from the campaign if roadmap is in approval or already approved
+	static testMethod void testRemoveCampaignRoadmapIfStageApprovedOrInApproval()
+	{
+		//create new roadmap
+		CW_Roadmap__c roadmap = TestDataUtil.getRoadmap();
+		insert roadmap;
+		
+		//create new campaign and link it to the roadmap
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+    	camp.Recurrence_Type__c = ConstantUtil.CAMPAIGN_RECURRENCE_TYPE_RECURRING;
+    	camp.Roadmap__c = roadmap.Id;
+    	insert camp;
+    	
+    	//approve roadmap
+    	roadmap.Approval_Status__c = 'Approved';
+    	update roadmap;
+    	
+    	//try to remove roadmap from campaign
+    	Test.startTest();
+    	camp.Roadmap__c = null;
+    	try 
+    		{
+    			update camp;
+    		}
+			catch (Exception e)
+			{
+				//return true if error message is displayed when trying to remove the roadmap from a campaign when roadmap approval status is approved or in approval 
+				Boolean expectedExceptionThrown =  e.getMessage().contains(ConstantUtil.REMOVE_ROADMAP_ERROR) ? true : false;
+				//error message should be dispalyed stoping the suer saving the campaign
+				System.AssertEquals(expectedExceptionThrown, true);
+			}
+    	
+    	Test.stopTest();
+	}
+	
+	//stop user update campaign points if roadmap was approved or is in approval
+	static testMethod void testChangePointsWhileRoadmapInApprovalOrApproved()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility